javascript/lib.c

```diff
@@ -52,10 +52,109 @@ napi_value FindAPI(napi_env env, napi_callback_info info) {
     return js_result;
 }
 
+size_t count_char(strzl_haystack_t strzl_haystack, char needle) {
+    size_t result = strzl_naive_count_char(strzl_haystack, needle);
+    return result;
+}
+
+napi_value CountSubstrAPI(napi_env env, napi_callback_info info) {
+    size_t argc = 3;
+    napi_value args[3];
+    napi_get_cb_info(env, info, &argc, args, NULL, NULL);
+
+    // Extract the C string from the JavaScript string for haystack and needle
+    size_t str_size;
```

Comment:
You don't need two separate values per string. Just keep the `size_t haystack_l, needle_l`.